README.md

```diff
@@ -1,2 +1,139 @@
+
+```sh
+export LOCALSTACK_API_KEY=<your-api-key>
+localstack start
+```
+
+> If you prefer running LocalStack in detached mode, you can add the `-d` flag to the `localstack start` command, and use Docker Desktop to view the logs.
+
+## Instructions
+
+You can build and deploy the sample application on LocalStack by running our `Makefile` commands. Run `make install` to install the dependencies and `make run` to create the infrastructure on LocalStack. Run `make stop` to delete the infrastructure by stopping LocalStack.
+
+Alternatively, here are instructions to deploy it manually step-by-step.
+
+### Deploy the infrastructure
+
+To deploy the infrastructure, we are using the Serverless Framework with the `serverless-localstack` plugin. The `serverless.yml` file contains the configuration for the infrastructure. The `serverless-localstack` plugin is configured to enable any requests to AWS to be redirected to a running LocalStack instance.
+
+Before deploying the infrastructure, we need to install the dependencies. Run the following commands to install the dependencies:
+
+```sh
+yarn
+```
+
+To deploy the infrastructure, run the following command:
+
+```sh
+SLS_DEBUG=1 yarn deploy
+```
+
+After a few seconds, the infrastructure should be deployed successfully. Fetch the API ID and API Key by running the following commands:
+
+```sh
+export APPSYNC_URL=http://localhost:4566/graphql
+export api_id=$(awslocal appsync list-graphql-apis | jq -r '(.graphqlApis[] | select(.name=="test-api")).apiId')
+api_key=$(awslocal appsync create-api-key --api-id $api_id | jq -r .apiKey.id)
+echo $api_key
+echo $api_id
+```
+
+### Setup a WebSocket connection
+
+To setup a WebSocket connection, we need to create a Python virtual environment and install the dependencies. Run the following commands to create a Python virtual environment and install the dependencies:
+
+```sh
+virtualenv .venv
+source .venv/bin/activate
+pip install -r requirements.txt
+```
+
+Next, we need to create a WebSocket connection to the AppSync GraphQL API. Run the following command to create a WebSocket connection:
+
+```sh
+python websocket_client.py $api_id
+```
+
+The above script will make use of the `wscat` utility for testing the AppSync GraphQL API. The script will create a WebSocket connection using the SDK for Python (Boto3) and start a subscription client. To disconnect from your API, enter **CTRL-C**.
+
+### Test the GraphQL API
+
+To test the GraphQL API, we will use `cURL` to invoke the integrations with DynamoDB and RDS. Let us send a mutation request to the AppSync API to add a new post to the DynamoDB data source. The mutation specifies the `addPostDDB` field and provides the `id` parameter with a value of `id123`. The response from the API will include the `id` of the newly created post.
+
+```sh
+curl -H "Content-Type: application/json" -H "x-api-key: $api_key" -d '{"query":"mutation {addPostDDB(id: \"id123\"){id}}"}' $APPSYNC_URL/$api_id
```

Comment:
nit: Would be great if we could add these commands to the `Makefile` as well, for easy testing (e.g., `make test-requests`). Doesn't have to be for this iteration, though - can also add that in a follow-up 👍 

Edit: just noticed that the commands are partially already present in `run.sh` below. Again, not critical - let's maybe add that in a follow-up iteration.